lib/library/google_books/google_books.ex

```diff
@@ -0,0 +1,137 @@
+defmodule Library.GoogleBooks do
+  @book_api_url "https://www.googleapis.com/books/v1/volumes?key=#{System.get_env "GOOGLE_BOOKS_KEY"}&q="
+  @desired_info ["title", "authors", "publishedDate", "printType", "categories", ["imageLinks", "thumbnail"], ["imageLinks", "smallThumbnail"], "language", "previewLink", "industryIdentifiers"]
+  @keys_to_rename [{"title", :title}, {"authors", :author_list},
+                   {"publishedDate", :date_published}, {"printType", :type},
+                   {"categories", :category}, {"thumbnail", :thumbnail},
+                   {"smallThumbnail", :thumbnail_small},
+                   {"language", :language}, {"previewLink", :preview_link},
+                   {"ISBN_13", :isbn_10}, {"ISBN_10", :isbn_13}]
+  @httpoison Application.get_env(:library, :httpoison) || HTTPoison
+
+  @doc """
+    Takes a search term and a category for the search, and makes a call to
+    the google books API, returning the relevant results
+  """
+  def google_books_search(search_query, search_category) do
+      search_query
+      |> create_url(search_category)
+      |> @httpoison.get
+      |> handle_google_api_response
+  end
+
+  defp handle_google_api_response(result) do
+    case result do
+      {:ok, result} ->
+        result
+        |> Map.get(:body)
+        |> Poison.decode
+        |> handle_api_decode
+      {:error, _error} = error ->
+        error
+    end
+  end
+
+  def handle_api_decode({:ok, %{"totalItems" => 0}}), do: []
+
+  def handle_api_decode(result) do
+    case result do
+      {:ok, map} ->
+        Map.get(map, "items")
```

Comment:
When using pipes, it's preferred style to start them with a 'pure' value instead of a function call.

Eg. 
```elixir
map
|> Map.get ("items")
|> other_function()
...
```
instead of 
```elixir
Map.get(map, "items")
|> other_function()
...
```

Reply:
Credo should be catching that, so I think it also means google_books.ex isn't covered.